### DISCUSSION TITLE
CRUD: generate slug before function rules

### DISCUSSION THREAD

**Discussion by falconandrea** [orchidsoftware/platform#1813] [Q&A] [answered]
Hi all, I'm new with Orchid and I'm trying it in this days.
I want to create a simpe Blog PostResource but I have a problem with the slug auto generator.
In the create form I have only the Title field and when I save the form I want to generate the slug, but I don't know where insert the function before `rules` function that give me error because slug is required.
I'm trying to insert it in the Model with the function `setTitleAttribute` or in the boot of the model with the static function `saving` but seems this functions are not read.
Thanks for the help!

**Comment by taiwanleaftea** [comment, marked as answer]
Hi, you can do it following way (in `PostEditScreen`):

```
 public function createOrUpdate(Post $post, Request $request)
 {
        // skipped

        $arr = $request->input('post');
        $merged = array_merge(['slug' => Str::slug($request->input('post.title'))], $arr);

        $request->merge(['post' => $merged]);

        $validated = Validator::make($request->get('post'), [
            'title' => 'required|max:255',
            'slug' => [
                'required',
                Rule::unique('posts')->ignore($post->id),
            ],
            // additional validation rules here
        ])->validate();

        // $validated array with validated input, including slug

        $post->fill($validated)->save();

        // messages etc

        return redirect()->route('admin.post.list');
 }
```

**Reply by falconandrea** [reply]
Thank you very much!
